src/evergreen/api.py

```diff
@@ -308,6 +309,18 @@ def project_by_id(self, project_id: str) -> Project:
         url = self._create_url("/projects/{project_id}".format(project_id=project_id))
         return Project(self._paginate(url), self)  # type: ignore[arg-type]
 
+    def project_by_name(self, project_name: str) -> Optional[Project]:
```

Comment:
`project_by_name` is a substitute for `project_by_id` that works for non-authorized Evergreen users (see this issue: https://jira.mongodb.org/browse/EVG-6878). I'm not sure what to name this `project_by_name` method.